feat(mainpage): new GeoGuessr main page

steve020607 opened the pull request:
## Summary
GeoGuessr New Main Page

## How did you test this change?
https://liquipedia.net/geoguessr/User:Steve23/Main_Page

steve020607 committed:
GeoGuessr New Main Page

hjpalpha reviewed (commented):
linter:

steve020607 committed:
Update MainPageLayoutData/Config

Co-authored-by: hjpalpha <75081997+hjpalpha@users.noreply.github.com>

steve020607 committed:
Add Filter Buttons for GeoGuessr

steve020607 committed:
Add GeoGuessr Banner

ElectricalBoy reviewed (commented):
lua files need to be moved to the new directories
see #5596

steve020607 commented:
> lua files need to be moved to the new directories see #5596
did you mean that i have to moved the New Main Page file to what folder? i'm not understand

steve020607 commented:
> > > lua files need to be moved to the new directories see #5596
> > 
> > 
> > did you mean that i have to moved the New Main Page file to what folder? i'm not understand
> 
> `components/filter_buttons/wikis/geoguessr/filter_buttons_config.lua` needs to go to `lua/wikis/geoguessr/filter_buttons/filter_buttons_config.lua`, and similar story applies to `components/main_page/wikis/geoguessr/main_page_layout_data.lua` too imo it may be better to wait until #5600 gets merged

did i have to change all the folder on my lua modules or i can just rename the file for main page?(since i see that my lua modules still using an old folder system)

https://github.com/steve020607/Lua-Modules/tree/patch-6

steve020607 committed:
Merge branch 'main' into patch-6

SlothyMan reviewed (commented):
missing content

steve020607 committed:
Update for the new revision on the module

Co-authored-by: SlothyMan <76825878+SlothyMan@users.noreply.github.com>

steve020607 committed:
Remove Help Navigation Card

SlothyMan commented on lua/wikis/geoguessr/main_page/main_page_layout_data.lua:
```diff
@@ -0,0 +1,216 @@
+		body = '{{Liquipedia:Want_to_help}}',
+		padding = true,
+		boxid = 1504,
+	},
```
```suggestion
	},
	liquipediaApp = {
		heading = 'Download the Liquipedia App',
		padding = true,
		body = '{{Liquipedia:App}}',
		boxid = 1505,
	},
```

steve020607 committed:
Add LiquipediaApp

SlothyMan commented on lua/wikis/geoguessr/main_page/main_page_layout_data.lua:
```diff
@@ -0,0 +1,216 @@
+					{
+						mobileOrder = 8,
+						content = CONTENT.wantToHelp,
+					},
```
```suggestion
					},
					{
						mobileOrder = 9,
						content = CONTENT.liquipediaApp,
					},
```

SlothyMan reviewed (commented):
~~liquipedia app content~~

steve020607 committed:
Update Main Page daya

Co-authored-by: SlothyMan <76825878+SlothyMan@users.noreply.github.com>

SlothyMan committed:
kick + fix

Kick app as I thought it must be on all new wikis.
kick content updates. Not defined and doesnt exist. (redundant)
Mobile order fixes

SlothyMan committed:
classic trailing spaces

SlothyMan committed:
Update main_page_layout_data.lua

SlothyMan committed:
Merge branch 'main' into patch-6

steve020607 committed:
Update main_page_layout_data.lua

steve020607 committed:
always this case that are make checker to be broke

steve020607 committed:
Merge branch 'main' into patch-6

ElectricalBoy reviewed (changes requested):
files should be renamed so that they match names on wiki (see #5600)

steve020607 committed:
Request by ElectricalBoy

steve020607 commented:
> files should be renamed so that they match names on wiki (see #5600)
make sure a little bit, moved it to this?
lua/wikis/geoguessr/MainPageLayout/data.lua

steve020607 commented:
and how the checker doesn't like the commit

steve020607 committed:
are this that make it broke

steve020607 commented on lua/wikis/geoguessr/main_page/main_page_layout_data.lua:
```diff
@@ -0,0 +1,212 @@
+				},
+				children = { Link{ children = 'See more matches', link = 'Liquipedia:Matches'} }
+			}
+		),
```
[1d8f303](https://github.com/Liquipedia/Lua-Modules/pull/5591/commits/1d8f3039b1882e47f0488c6b50868b132448e9ac)
done

steve020607 committed:
Change name

ElectricalBoy reviewed (changes requested):
- `lua/wikis/geoguessr/filter_buttons/filter_buttons_config.lua` should be `lua/wikis/geoguessr/FilterButtons/Config.lua`
- `lua/wikis/geoguessr/main_page/data.lua` should be `lua/wikis/geoguessr/MainPageLayout/data.lua`

steve020607 committed:
Rename MainPageLayout file

steve020607 committed:
Change Filter name

steve020607 committed:
Update MainPage

steve020607 commented on lua/wikis/geoguessr/main_page/data.lua:
```diff
@@ -0,0 +1,197 @@
+		body = MatchTicker{},
+		padding = true,
+		boxid = 1507,
+	},
```
[1b2e49d](https://github.com/Liquipedia/Lua-Modules/pull/5591/commits/1b2e49d32914113c5abfdd6d89765f53c65887ee) [d906a3c](https://github.com/Liquipedia/Lua-Modules/pull/5591/commits/d906a3c492b35c0580111571bed372bf8e390018) (rename)
[6afc594](https://github.com/Liquipedia/Lua-Modules/pull/5591/commits/6afc5945aa132a76852717bc6f72a07247ee94f9) (add panelAttributes)

hjpalpha reviewed (approved):
seems okay on phone

steve020607 committed:
Change format navigation card to the Game Modes navigation card
